python/lsst/meas/extensions/shapeHSM/_hsm_moments.py

```diff
@@ -0,0 +1,548 @@
+
+        # Turn bounding box corners into GalSim bounds.
+        xmin, xmax = bbox.getMinX(), bbox.getMaxX()
+        ymin, ymax = bbox.getMinY(), bbox.getMaxY()
+        bounds = galsim.bounds.BoundsI(xmin, xmax, ymin, ymax)
+
+        # Get the `lsst.meas.base` mask for bad pixels.
+        badpix = exposure[bbox].mask.array.copy()
+        bitValue = exposure.mask.getPlaneBitMask(self.config.badMaskPlanes)
+        badpix &= bitValue
+
+        # Extract the numpy array underlying the image within the bounding box
+        # of the source.
+        imageArray = exposure[bbox].getImage().array
+
+        # Create a GalSim image using the extracted array.
+        # NOTE: GalSim's HSM uses the FITS convention of 1,1 for the
+        # lower-left corner.
+        image = galsim.Image(imageArray, bounds=bounds, copy=False)
+
+        # Convert the mask of bad pixels to a format suitable for galsim.
+        # NOTE: galsim.Image will match whatever dtype the input array is
+        # (here int32).
+        badpix = galsim.Image(badpix, bounds=bounds)
+
+        # Call the internal method to calculate adaptive moments using GalSim.
+        self._calculate(
+            record,
+            image=image,
+            weight=None,
+            badpix=badpix,
+            sigma=2.5 * psfSigma,
+            precision=1.0e-6,
+            centroid=center,
+        )
+
+
+class HsmSourceMomentsRoundConfig(HsmSourceMomentsConfig):
+    """Configuration for HSM adaptive moments measurement for sources using
+    round weight function.
+    """
+
+    def setDefaults(self):
+        super().setDefaults()
+        self.roundMoments = True
+
+    def validate(self):
+        if not self.roundMoments:
+            raise pexConfig.FieldValidationError(
+                self.roundMoments, self, "roundMoments should be set to `True`."
+            )
+        super().validate()
+
+
+@measBase.register("ext_shapeHSM_HsmSourceMomentsRound")
+class HsmSourceMomentsRoundPlugin(HsmSourceMomentsPlugin):
+    """Plugin for HSM adaptive moments measurement for sources using round
+    weight function.
+    """
+
+    ConfigClass = HsmSourceMomentsRoundConfig
+
+
+class HsmPsfMomentsConfig(HsmMomentsConfig):
+    """Configuration for HSM adaptive moments measurement for PSFs."""
+
+    useSourceCentroidOffset = pexConfig.Field[bool](doc="Use source centroid offset?", default=False)
+
+    def setDefaults(self):
+        super().setDefaults()
+        self.subtractCenter = True
+
+
+@measBase.register("ext_shapeHSM_HsmPsfMoments")
+class HsmPsfMomentsPlugin(HsmMomentsPlugin):
+    """Plugin for HSM adaptive moments measurement for PSFs."""
+
+    ConfigClass = HsmPsfMomentsConfig
+
+    def __init__(self, config, name, schema, metadata, logName=None):
+        super().__init__(config, name, schema, metadata, logName=logName)
+        docPrefix = "Debiased centroid" if "Debiased" in self.__class__.__name__ else "Centroid"
```

Comment:
Instead of this, define a private class variable, say `_debiased`, which is `False` for this one, and `True` for the debiased plugin. Condition the `docPrefix` based on that class variable instead of relying on the class name.

Reply:
And if you do that, you could pass that value for `debiasedPsfMoments` in the call to `_measure` and avoid overriding `measure` in `HsmPsfMomentsDebiasedPlugin`.

Reply:
Done! It simplified even more things.